### PR TITLE
16 json logical types

### PR DESCRIPTION
**Pull request by azurecoder**
Fixes #

Fixed where the isset is pulling a false value and failing on a branch issues with strings and byte array conversions as a result

### Description

This is in the wrong feature branch - it should be #54 - this is the first PR of several changes are:
- Object array returned from Parquet Dataset
- Indexer support on dataset
- Changes to parq cmd line tool to display contents of files 

### TODO

- Complete a new row to allow for paging and distribution 
- Make Parquet Dataset a paged model and DataFrame as an abstract model

### CODE REVIEW

**Comment by azurecoder** [comment]
removed the TODO - just a Richard reminder not a todo as such

File: src/Parquet/ParquetReader.cs [hunk 1]
```diff
@@ -119,6 +119,7 @@ private FileMetaData ReadMetadata()
 
       public void Dispose()
       {
+         _input.Dispose();
```
Comment (azurecoder) [inline src/Parquet/ParquetReader.cs]: good point, I'll update this

File: src/Parquet/ParquetDataSet.cs [hunk 2]
```diff
@@ -68,12 +70,18 @@ public ParquetDataSet(params ParquetColumn[] columns) : this((IEnumerable<Parque
...
-         {
-            return Columns.Select(column => column.Values[row]).ToArray();
-         }
-      }
+      /// <summary>
+      /// Used to get the row values
+      /// </summary>
+      /// <param name="row">The index of the row beginning with zero</param>
+      /// <returns>An object array</returns>
+      public object[] this[int row] => Columns.Select(column => column.Values[row]).ToArray();
```
Comment (azurecoder) [inline src/Parquet/ParquetDataSet.cs]: yeah I agree I want the whole Dataset, DataFrame model to change and to be inline with pandas, we need a session to flush this out properly

**Comment by azurecoder** [comment]
@aloneguid have updated with recommendations please square :-)

**Comment by aloneguid** [comment]
thank you )

**Pull request merged**